gateways/custom/storage-grid/main.go

```diff
@@ -61,26 +66,70 @@ type storageGridEventConfig struct {
 	Endpoint string
 	// Todo: add event and prefix filtering.
 	Events []string
-	Filter StorageGridFilter
+	Filter *Filter
 	// srv holds reference to http server
 	srv *http.Server
 	mux *http.ServeMux
 }
 
+// Filter represents filters to apply to bucket nofifications for specifying constraints on objects
+type Filter struct {
```

Comment:
been thinking more about this kind of stuff.. the user needs to know the structure of all these types when he/she builds the appropriate configmap for a gateway.. don't you think this belongs in the API package somewhere? But at the very least, how do we ensure this stuff is transparent, and validated against whenever a user creates a config?

Reply:
I agree that this needs to be transparent. We can add `Validate` method to `ConfigExecutor` interface to make sure that configs are validated.

Reply:
Also a separate `types.go` file containing structs might help with readability